### DISCUSSION TITLE
Is there a GUI design app for the Custom Tkinter?

### DISCUSSION THREAD

**Discussion by giorgiosld** [TomSchimansky/CustomTkinter#1316] [Q&A]
Hi i'm a newbie using Custom Tkinter, and i was wondering if there is any tool that makes creating GUI easier than just wiriting code. Thank to anyone who will answer! :)

**Comment by alfmelchor** [comment]
Not that I know of. However, I would recommend you look into Tkinter and CustomTkinter. I learned the library(s) in a few days and they're so fun and easy to develop. Plus, being able to develop from the ground up gives you the freedom to change every aspect of your GUI, unlike using a GUI designer. Best of luck!

**Reply by giorgiosld** [reply]
Thanks so much for the suggestion! :))

**Reply by MaxNiark** [reply]
i can share this for you 

for my part this is the best way for create a big app ;) Each widget needed to be class object, and u import the class in main frame widget 

import the CTkFrame in other .py as Main_APP :) . 

```
import tkinter as tk
from tkinter import ttk
from customtkinter import *



 ############## MAIN FRAME ##################
class Dash_board(CTkFrame):
  def __init__(self,list_db_lic,*args, **kwargs):
    super().__init__(*args, **kwargs)

    Frame_dashboard_total=CTkScrollableFrame(self,label_text="dashboard",bg_color="black",fg_color="purple",corner_radius=20, width=200, height=500)
    Frame_dashboard_total.grid(pady=10,padx=10,row=0,column=0, sticky="")

    frame_list_of_projet=list_of_projet(width=600,master=self)   ## import class object widget 
    frame_list_of_projet.grid(row=0, column=1)

    tabview=CTkTabview(Frame_dashboard_total)
    tabview.grid()

    tab_1 = tabview.add("tab 1")
    tab_1.grid()

    

    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid(column=1)
    label=CTkLabel(tab_1)
    label.grid(column=1)
    label=CTkLabel(tab_1)
    label.grid(column=1)
    label=CTkLabel(tab_1)
    label.grid(column=1)
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(tab_1)
    label.grid(column=1)
    label=CTkLabel(tab_1)
    label.grid()
    label=CTkLabel(Frame_dashboard_total)
    label.grid(column=0)
    label=CTkLabel(Frame_dashboard_total)
    label.grid(column=1)
    label=CTkLabel(Frame_dashboard_total)
    label.grid(column=2)
    label=CTkLabel(Frame_dashboard_total)
    label.grid(column=3)
    label=CTkLabel(Frame_dashboard_total)
   # label.grid(column=4)
    label=CTkLabel(Frame_dashboard_total)
   # label.grid(column=5)
    label=CTkLabel(Frame_dashboard_total)
   # label.grid(column=6)
    label=CTkLabel(Frame_dashboard_total)
   # label.grid(column=7)
    label=CTkLabel(Frame_dashboard_total)
   # label.grid(column=8)
   # label=CTkLabel(Frame_dashboard_total)
  #  label.grid(column=9)
  #  label=CTkLabel(Frame_dashboard_total)
  #  label.grid(column=10)

          ################ IN MAIN FRAME #############################
class list_of_projet(CTkScrollableFrame):
    def __init__(self,*args, **kwargs):
        super().__init__(*args, **kwargs)
  
        Frame_list_projet=ttk.Treeview(self)
        Frame_list_projet.grid()


```

**Comment by Akascape** [comment]
There is a GUI design app I made:
CTkDesigner: https://ko-fi.com/s/6fca1ae70f

https://github.com/Akascape/CTkDesigner-Support

[![Screenshot](https://github.com/Akascape/CTkDesigner-Support/assets/89206401/6435f49b-f7d0-4cba-8190-73cd71d77ac3)](https://ko-fi.com/s/6fca1ae70f)

## Features
- DRAG & DROP
- Move widgets
- Adjust all parameters
- Precise place method
- 27 CTk widgets 
- Create and preview themes
- Export app to .py
- Save/Open created templates
- All required functionalities and shortcuts added
- WYSIWYG

**Reply by ghost** [reply]
There is Pygubu Designer. It looks good ehen using tkinter and can use CTk objects as well. But... it doesn't generate the right code, unfortunately .

**Reply by Akascape** [reply]
@FrankteWH But ctkdesigner is going to be more advanced.
I added lots of new widget which are made specially for customtkinter.
Moreover, it saves created ui as template, generates correct .py code.

![Screenshot 2024-01-12 212235](https://github.com/TomSchimansky/CustomTkinter/assets/89206401/847a1e0b-39aa-475e-8546-6346c3c00a16)

**Reply by DhinagaranDhina** [reply]
@Akascape, i need this app. i will do Gpay if you are from india.

**Reply by Akascape** [reply]
@DhinagaranDhina I will be moving this to a better platform where you can pay with indian payment options. Don't worry. Wait for version 2. 
I will notify you.

**Reply by DhinagaranDhina** [reply]
@Akascape - when can i expects V2.0?
Please do some improvement in CTkTableview.  you can consider tableview widgets from ttkbootstrap package for improvement.

**Reply by Akascape** [reply]
Next month, ctktabview can be improved but I will add it later (maybe version3), because it need lots of work.

**Reply by Juloup78** [reply]
will the v2 be free for users that paid for v1

**Reply by Akascape** [reply]
@Juloup78 Yes, but for new users, price may vary.

**Reply by DhinagaranDhina** [reply]
@Akascape v2 released ?

**Comment by scaramacai** [comment]
[pygubu-designer](https://github.com/alejandroautalan/pygubu-designer) has good support for CTk.

**Reply by ghost** [reply]
Which version do you  use? Mine from 7 januari 2024 is buggy. It experts a CtkFrame.name potmeter that doesn't experts....🤔😪

**Reply by GRKSEO** [reply]
I've faced similar issues with Tkinter recently! The version I used had trouble rendering CtkFrame properly, especially with nested widgets. It felt like some of the features were half-baked, and debugging was a headache. 😪

If you're looking for an easier way to design GUIs with Tkinter, you might want to check out this [Custom Tkinter GUI Designer](https://labdeck.com/python-designer/custom-tkinter-gui-designer/). It simplifies a lot of the manual work and helps avoid some of these common bugs. Might be worth a try!

**Comment by scaramacai** [comment]
I'm using pygubu-designer 0.37 together with pygubu 0.32. There are issues that I already opened on github and the generated code is far from being perfect but it is mostly usable and a good starting point for designing also a complex interface. I must say, however, that I didn't test too much the pygugu builder, but mainly the generated script, using python directly.

**Comment by Proxlight** [comment]
> Hi i'm a newbie using Custom Tkinter, and i was wondering if there is any tool that makes creating GUI easier than just wiriting code. Thank to anyone who will answer! :)

If you're looking to design a modern GUI in Python, I highly recommend checking out Buildfy GUI! It’s a user-friendly desktop app built with CustomTkinter, offering a sleek, responsive interface for creating, testing, and deploying web applications directly from your desktop. It provides features like real-time code previews, project management, and one-click deployment—all with an intuitive, modern design.

You can try it out here: https://buildfygui.netlify.app/

It’s a great option if you want a polished, professional GUI without dealing with too much complexity!

**Comment by rigvedmaanas** [comment]
> Hi i'm a newbie using Custom Tkinter, and i was wondering if there is any tool that makes creating GUI easier than just wiriting code. Thank to anyone who will answer! :)

I created Custom Tkinter Builder to design UI by just drag and dropping widgets.

This is a tool for anyone who wants to build GUIs for Python projects without needing to write tons of code. Instead of writing everything out, you can just drag and drop widgets and customize them with this software and generate the necessary code.
You can change the properties of all the widgets according to your preferences 🤩. This software has builtin Icon support which contains more than **2000** **icons** with **varied color**, size and type from **Material Design Icons**. This software will help you to create responsive and better looking UI faster than ever ⚡. It’s **free and cross-platform** too.

This is the how the software looks.
![software](https://github.com/user-attachments/assets/4dc7eaa2-cb26-4385-96b0-ec6341a1d0c7)

Repo: [https://github.com/rigvedmaanas/CustomTkinterBuilder](https://github.com/rigvedmaanas/CustomTkinterBuilder)

_Built for Custom Tkinter using Custom Tkinter_
**Over 170 stars 🚀**

**Comment by Proxlight** [comment]
> Hi i'm a newbie using Custom Tkinter, and i was wondering if there is any tool that makes creating GUI easier than just wiriting code. Thank to anyone who will answer! :)

Hi there! 👋 Welcome to the world of CustomTkinter! 😊 If you're looking for a tool to simplify GUI creation, I highly recommend trying Buildfy.

![001](https://github.com/user-attachments/assets/3deede53-ff14-4880-83b3-5a5dd6e8d488)


It's a drag-and-drop GUI builder specifically designed for Python's CustomTkinter. With Buildfy, you can visually design your interfaces, preview them in real-time, and even export the layout as clean Python code—saving you the hassle of manual coding. 🎨💻

You can check out the free version here: [Buildfy](https://buildfygui.netlify.app), and if you're interested in more advanced features, the Pro version is available too!

Feel free to reach out if you need help getting started. Happy coding! 🚀

**Comment by Proxlight** [comment]
I created Buildfy, an AI-powered drag-and-drop GUI builder for Python, designed to make creating stunning UIs fast and easy. With Buildfy, you don’t need to write lines of code. Just drag and drop widgets and let the AI generate the necessary code for you.

All the widgets are available for free users, and you can customize them to fit your project’s needs with ease 🎨. Buildfy even comes with AI-assisted design features that help you craft beautiful and modern UIs quickly, without coding knowledge.

No more struggling with complex code—just design your GUI visually, and Buildfy will handle the rest ⚡.
![Generative AI](https://github.com/user-attachments/assets/d0c80738-883b-428f-ac57-dbf913a57d93)


Key Features:

Drag and drop interface design
Access to all widgets (free!)
AI-powered GUI creation
Cross-platform and free to use
Download and start building stunning Python GUIs: https://buildfygui.netlify.app/
Watch the full tutorial: https://youtu.be/h65qzqee4nU?si=8Ys9GU-zTyxgRFKh

#Python #PythonGUI #AI #Buildfy #Tkinter #AppDevelopment #NoCode
